tests/testthat/test-interlace.R

```diff
@@ -34,11 +34,22 @@ test_that("interlace.tbl_sql() works", {
     )
 
 
-    expect_identical(interlace(list(t1, t2), by = "key") |> dplyr::collect(),
-                     t_ref |> dplyr::collect())
-
-    expect_mapequal(interlace(list(t1, t2), by = "key") |> dplyr::collect(),
-                    interlace(list(t2, t1), by = "key") |> dplyr::collect())
+    # Order of records may be different, so we arrange then check if they are identical
+    expect_identical(interlace(list(t1, t2), by = "key") |>
+                       dplyr::collect() |>
+                       dplyr::arrange(.data$key, .data$valid_from),
+                     t_ref |>
+                       dplyr::collect() |>
+                       dplyr::arrange(.data$key, .data$valid_from))
+
+    # Order of columns will be different, so we only require a mapequal
+    # .. but order of records can still be different
+    expect_mapequal(interlace(list(t1, t2), by = "key") |>
```

Comment:
I assume that t1 and t2 should be swapped in the second test.

Reply:
Actually, this test is to check that the order of the arguments does not change (functionally) change the output.
The "data" of either call should be identical, but column and record order can be permuted. We only require that they contain equivalent data.

maprequal ignores column order and by sorting, we ignore differences in record order.

Reply:
Understood